src/typed-error.ts

```diff
@@ -19,6 +19,7 @@ export class TypedError extends Error {
 	public name: string;
 	public message: string;
 	public stack: string = '';
+	public payload: any | null;
```

Comment:
We don't just use this module in the API - it's also used in the SDK and various other places, and 'payload' is a bit specific. `metadata`? That's effectively what this is I think - extra data about the error that downstream users might find useful (and perhaps might want to include in an HTTP response).